sdks/python/apache_beam/io/gcp/pubsub.py

```diff
@@ -36,16 +38,122 @@
 from apache_beam.transforms import window
 from apache_beam.transforms.display import DisplayDataItem
 
-__all__ = ['ReadStringsFromPubSub', 'WriteStringsToPubSub']
+__all__ = ['PubsubMessage', 'ReadMessagesFromPubSub', 'ReadStringsFromPubSub',
+           'WriteStringsToPubSub']
+
+
+class PubsubMessage(object):
+  """Represents a message from Cloud Pub/Sub.
+
+  This interface is experimental. No backwards compatibility guarantees.
+  """
+
+  def __init__(self, payload, message_id, attributes, publish_time):
+    """Constructs a message.
+
+    This interface is experimental. No backwards compatibility guarantees.
+
+    See also:
+      https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage
+
+    Attributes:
+      payload: (str) Message payload, unencoded.
```

Comment:
<!--new_thread; commit:6bfe8d93b2cb863f16655addcbb98b35d64ce365; resolved:0-->
What does "unencoded" mean?  Do you mean raw bytes as opposed to unicode?

Reply:
Yes. Not unicode and not base64 (unlike https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage). Not sure of a better way to convey that.

Reply:
There is an ambiguity in the way this is phrased--if you have a unicode string, you can call `ustr.encode('utf-8')` to get a byte string with the unicode characters encoded into bytes using the utf-8 codec (or utf-16, etc).  In this way, the byte string would be considered the unicode string  "encoded" with the codec.  Since this is confusing, can you just say "Message payload, as a byte string"?